server/protocols/webadmin/webadmin-mailbox/src/test/java/org/apache/james/webadmin/routes/EventDeadLettersRoutesTest.java

```diff
@@ -327,12 +344,460 @@ void deleteShouldRemoveEvent() {
+                .get("/events/deadLetter/groups/" + SERIALIZED_GROUP_A + "/" + UUID_1)
+            .then()
+                .statusCode(HttpStatus.NOT_FOUND_404);
+        }
+
+        @Test
+        void postRedeliverAllEventsShouldRedeliverAndRemoveAllEventsFromDeadLetters() {
+            deadLetters.store(groupA, EVENT_1).block();
+            deadLetters.store(groupA, EVENT_2).block();
+            deadLetters.store(groupB, EVENT_2).block();
+
+            String taskId = with()
+                .queryParam("action", EVENTS_ACTION)
+                .post("/events/deadLetter")
+                .jsonPath()
+                .get("taskId");
+
+            given()
+                .basePath(TasksRoutes.BASE)
+            .when()
+                .get(taskId + "/await")
+            .then()
+                .body("status", is("completed"))
+                .body("additionalInformation.successfulRedeliveriesCount", is(3))
+                .body("additionalInformation.failedRedeliveriesCount", is(0));
+
+            when()
+                .get("/events/deadLetter/groups/" + SERIALIZED_GROUP_A + "/" + UUID_1)
+            .then()
+                .statusCode(HttpStatus.NOT_FOUND_404);
+
+            when()
+                .get("/events/deadLetter/groups/" + SERIALIZED_GROUP_A + "/" + UUID_2)
+            .then()
+                .statusCode(HttpStatus.NOT_FOUND_404);
+
+            when()
+                .get("/events/deadLetter/groups/" + SERIALIZED_GROUP_B + "/" + UUID_2)
+            .then()
+                .statusCode(HttpStatus.NOT_FOUND_404);
+        }
+
+        @Test
+        void postRedeliverAllEventsShouldFailWhenInvalidAction() {
+            deadLetters.store(groupA, EVENT_1).block();
+
+            given()
+                .queryParam("action", "invalid-action")
+            .when()
+                .post("/events/deadLetter")
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(ContentType.JSON)
+                .body("statusCode", is(400))
+                .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+                .body("message", is("Invalid arguments supplied in the user request"))
+                .body("details", is("invalid-action is not a supported action"));
+        }
+
+        @Test
+        void postRedeliverAllEventsShouldFailWhenMissingAction() {
+            deadLetters.store(new EventBusTestFixture.GroupA(), EVENT_1).block();
+
+            when()
+                .post("/events/deadLetter")
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(ContentType.JSON)
+                .body("statusCode", is(400))
+                .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+                .body("message", is("Invalid arguments supplied in the user request"))
+                .body("details", is("'action' url parameter is mandatory"));
+        }
```

Comment:
I miss a test asserting that `re-deliver is called.`

Suggestion:

 - 1. Register an EventCollector on group A
 - 2. Call reDeliver action
 - 3. Await that there is an event in the eventCollector